File: practice/src/main/java/com/decade/practice/security/TokenCredentialService.java
package com.decade.practice.security;

import com.decade.practice.model.TokenCredential;
import com.decade.practice.model.domain.entity.User;
import com.decade.practice.security.model.UserClaims;
import org.springframework.security.access.AccessDeniedException;

import java.util.List;

public interface TokenCredentialService {

        void validate(String refreshToken) throws AccessDeniedException;

        List<String> evict(String username);

        void evict(String username, String refreshToken);

        void add(String username, String refreshToken);

        UserClaims decodeToken(String token);

        TokenCredential create(UserClaims claims, String refreshToken);

        TokenCredential create(User user, String refreshToken);

        List<String> get(String username);
}

File: practice/src/main/java/com/decade/practice/security/TokenCredentialService.kt
package com.decade.practice.security

import com.decade.practice.model.entity.User
import com.decade.practice.security.model.TokenCredential
import com.decade.practice.security.model.UserClaims
import org.springframework.security.access.AccessDeniedException

interface TokenCredentialService {

      @Throws(AccessDeniedException::class)
      fun validate(refreshToken: String)
      fun evict(username: String): List<String>
      fun add(username: String, refreshToken: String)

      fun decodeToken(token: String): UserClaims
      fun create(claims: UserClaims, refreshToken: String? = null): TokenCredential
      fun create(user: User, refreshToken: String? = null): TokenCredential
      fun get(username: String): List<String>
}